File: Indexed Search/Hash Based Indexed Search/Perfect Hashing/PerfectHashing.py
class PerfectHashing:

    def __init__(self):

        self.data = set()

    def build(
        self,
        keys
    ):

        for key in keys:
            self.data.add(
                key
            )

    def search(
        self,
        key
    ):

        return (
            key
            in
            self.data
        )


table = PerfectHashing()

table.build(
    [
        10,
        25,
        42,
        60
    ]
)

print(
    table.search(
        42
    )
)

File: Indexed Search/Hash Based Indexed Search/Perfect Hashing/PerfectHashing.js
class PerfectHashing {

    constructor() {

        this.table =
            new Set();
    }

    build(
        keys
    ) {

        keys.forEach(
            key =>
            this.table
            .add(
                key
            )
        );
    }

    search(
        key
    ) {

        return (
            this.table
            .has(
                key
            )
        );
    }
}

const hash =
new PerfectHashing();

hash.build(
[
10,
25,
42
]
);

console.log(
hash.search(
42
)
);
